eclib/JArray.ec

```diff
@@ -523,3 +523,28 @@ abstract theory PolyArray.
 
 
 end PolyArray.
+
+(* Array slicing *)
+abstract theory SubArray.
+  op sizeS: int.
+  axiom gt0_sizeS: 0 < sizeS.
+
+  op sizeB: int.
+  axiom gt0_sizeB: 0 < sizeB.
+
+  (* Sub-array *)
+  clone import PolyArray as ArrayS with
+    op size <- sizeS
+    proof ge0_size by rewrite ltzW gt0_sizeS.
+
+  (* Base array *)
+  clone import PolyArray as ArrayB with
+    op size <- sizeB
```

Comment:
same comment.

Reply:
The semantic is the following
clone T as T'
   x <- t     :    x will be substituted by t everywhere in T', and its definition is removed so T'.x does not exists
   x = t        : T'.x is defined to be t, all occurences of x in T' is unchanged.
   x <= t     : T'.x is defined to be t,  x will be substituted by t everywhere in T'.